.circleci/config.yml

```diff
@@ -577,21 +628,69 @@ jobs:
+            cd ~/generated-sql
+
+            # Commits on generated-sql are tagged with the sha1 hash of the commit on main that triggered the SQL generation
+            export LATEST_TAG_GENERATED_SQL=`git describe --tags --abbrev=0`
+            LATEST_TAG_GENERATED_SQL=${LATEST_TAG_GENERATED_SQL/c-/""}
+
+            cd ~/project
+            export GENERATED_SQL_CHANGED=false
+
+            # If the between main the generated-sql branch SQL generators have changed, then re-run the SQL generation
+            if [[ $(git diff --name-only --diff-filter=d $LATEST_TAG_GENERATED_SQL...HEAD -- sql_generators) ]]; then
+              echo "SQL generator changed; re-generate SQL"
+              GENERATED_SQL_CHANGED=true
+            fi
+
+            # If a SQL was removed from main re-run SQL generation
+            # simply merging the changes with the generated-sql branch would result in files being part of generated DAGs
+            # and other artifacts that were removed
+            if [[ $(git diff --name-only --diff-filter=D $LATEST_TAG_GENERATED_SQL...HEAD -- sql) ]]; then
+              echo "SQL file was removed; re-generate SQL"
+              GENERATED_SQL_CHANGED=true
+            fi
+
+            mkdir -p /tmp/workspace/main-generated-sql
+            mkdir -p /tmp/workspace/main-generated-sql/dags
+
+            if [ "$GENERATED_SQL_CHANGED" = "true" ]; then
+              # re-run SQL generation and re-generate DAGs
+              ./script/bqetl generate all \
+                --target-project moz-fx-data-shared-prod
+
+              cp -r sql/ /tmp/workspace/main-generated-sql/sql
+            else
+              echo "Changes made don't affect generated SQL. Use content from generated-sql"
+
+              # merge files that changed in this PR with the generated-sql branch
+              cp -r ~/generated-sql/sql/. /tmp/workspace/main-generated-sql/sql
+              cp -a sql/. /tmp/workspace/main-generated-sql/sql
+            fi
+
             ./script/bqetl query render \
-              --output-dir sql/ \
-              sql/
+              --sql-dir /tmp/workspace/main-generated-sql/sql/ \
+              --output-dir /tmp/workspace/main-generated-sql/sql/ \
+              /tmp/workspace/main-generated-sql/sql/
             ./script/bqetl dependency record \
               --skip-existing \
-              "sql/"
+              "/tmp/workspace/main-generated-sql/sql/"
             ./script/bqetl metadata update \
-              sql/
+              --sql-dir /tmp/workspace/main-generated-sql/sql/ \
+              /tmp/workspace/main-generated-sql/sql/
+            # re-generate DAGs
 
-            mkdir -p /tmp/workspace/main-generated-sql
-            cp -r sql/ /tmp/workspace/main-generated-sql/sql
-            mkdir -p /tmp/workspace/main-generated-sql/dags
-            ./script/bqetl dag generate --output-dir /tmp/workspace/main-generated-sql/dags
+            ./script/bqetl dag generate \
```

Comment:
Could you copy the DAGs too?

Reply:
The DAGs need to be re-generated if there is any newly scheduled query, or a query references a new table. So I think DAG generations happen in a lot of PRs. And luckily the DAG generation is pretty quick.